daemon/src/main/scala/org/apache/gearpump/cluster/master/AppManager.scala

```diff
@@ -310,22 +320,20 @@ private[cluster] class AppManager(kvService: ActorRef, launcher: AppMasterLaunch
   case class RecoverApplication(applicationStatus: ApplicationState)
 
   private def cleanApplicationData(appId: Int): Unit = {
-    // Add the dead app to dead appMaster
-    appMasterRegistry.get(appId).foreach { pair =>
-      val (appMasterActor, info) = pair
-      deadAppMasters += appId -> (appMasterActor, info.copy(
-        finishTime = System.currentTimeMillis()))
-    }
-
-    appMasterRegistry -= appId
+    // Add the dead app to dead appMasters
+    deadAppMasters += appId
```

Comment:
When the application is down, you don't update its finish time. You can see the dashboard missed that info.